CRM/Core/Smarty.php

```diff
@@ -204,13 +202,47 @@ private function initialize() {
       $this->registerPlugin('modifier', $function, $function);
     }
 
+    $this->registerPlugin('modifier', 'call_user_func', [self::class, 'callUserFuncArray']);
+    // This does not appear to be used & feels like the sort of approach that would be phased out.
     $this->assign('crmPermissions', new CRM_Core_Smarty_Permissions());
 
     if ($config->debug || str_contains(CIVICRM_UF_BASEURL, 'localhost') || CRM_Utils_Constant::value('CIVICRM_UF') === 'UnitTests') {
       $this->error_reporting = E_ALL;
     }
   }
 
+  /**
+   * Call a permitted function from the Smarty layer.
+   *
+   * In general calling functions from the Smarty layer is being made stricter in
+   * Smarty - they need to be registered.
+   *
+   * We can't quite kill off call_user_func from the smarty layer yet but we
+   * can deprecate using it to call anything other than the 3 known patterns.
+   * In Smarty5 this will hard-fail, which is OK as Smarty5 is being phased in
+   * and can err on the side of strictness, at least for now.
+   *
+   * @param callable $callable
+   * @param mixed $args
+   *
+   * @return mixed
+   * @throws \CRM_Core_Exception
+   */
+  public static function callUserFuncArray(callable $callable, ...$args) {
+    $permitted = [
+      ['CRM_Campaign_BAO_Campaign', 'isComponentEnabled'],
+      ['CRM_Case_BAO_Case', 'checkPermission'],
+      ['CRM_Core_Permission', 'check'],
+    ];
+    if (!in_array($callable, $permitted)) {
+      if (CRM_Core_Smarty::singleton()->getVersion() === 5) {
+        throw new CRM_Core_Exception('unsupported function');
```

Comment:
Hmm every page crashes now if you have the fulltext search block in the sidebar. I know that's deprecated, but my general thought is it's difficult to track down why it's crashing because you don't get any information. Even with the deprecated warning in v4, it floods your logs but doesn't tell you where it is.

Is there an easy way to see which template is calling this? Adding a `CRM_Core_Error::backtrace('msg', true)` helps a little.